### ISSUE TITLE
Version 2.0.3 broke jshint config loading in some way. 2.0.2 works.

### ISSUE THREAD

**Issue by estreen** [spalger/gulp-jshint#152]
Starting with version 2.0.3 a build that previously worked started to fail because globals were no longer defined. JSHint started complaining about undefined 'angular' and '_' in this particular case. Likely due to the dependency update of rcloader between 2.0.2 and 2.0.3.

Downgrading back to 2.0.2 makes it work again.

**Comment by vankasteelj** [comment]
The update 2.0.3 basically breaks my workflow (and my CI), because I use pre-commit hooks & jshint on `npm test`.

**Comment by spalger** [comment]
Can you please share your configs?

**Comment by estreen** [comment]
Took a little while to come up with a small reproducer. The issue is a bit weirder than just something going wrong with the config.

In the attached reproducer the "lint" task works if gulp-jshint is downgraded to 2.0.2, strangely enough it will work with 2.0.3 if the essentially empty file src/app.js is removed. The errors are all in src/test/index.js. I verified that it is still getting linted when app.js is not present.

Running jshint manually on the two files also works.

Hope this can help track it down. Took a while to isolate out a small reproducer from a larger app/build system.

[repro.zip](https://github.com/spalger/gulp-jshint/files/589098/repro.zip)

**Comment by DarthVanger** [comment]
Having the same issue, getting warnings about `undefined` `'angular'`, `'$'`, etc, after upgrading to `2.0.3`:
```
12:23:54.491 ../webapp/resources/js/constants.js
12:23:54.491   line 7  col 1  'define' is not defined.
12:23:54.491 
12:23:54.491   ⚠  1 warning
12:23:54.491 
12:23:54.528 
12:23:54.528 ../webapp/resources/js/constants.min.js
12:23:54.528   line 1  col 1  'define' is not defined.
12:23:54.528 
12:23:54.528   ⚠  1 warning
12:23:54.528 
12:23:54.702 
12:23:54.702 ../webapp/resources/js/directives.js
12:23:54.702   line 1    col 1   'define' is not defined.
12:23:54.702   line 8    col 17  '$' is not defined.
12:23:54.702   line 62   col 21  '$' is not defined.
12:23:54.702   line 139  col 33  '$' is not defined.
12:23:54.702   line 141  col 39  '$' is not defined.
12:23:54.702   line 166  col 25  '$' is not defined.
12:23:54.702   line 181  col 13  '$' is not defined.
....
```
My `.jshintrc`:
```
$ cat .jshintrc
{
    "bitwise": true,
    "camelcase": true,
    "curly": true,
    "eqeqeq": true,
    "esversion": 6,
    "forin": true,
    "freeze": true,
    "immed": true,
    "indent": 4,
    "latedef": "nofunc",
    "newcap": true,
    "noarg": true,
    "noempty": true,
    "nonbsp": true,
    "nonew": true,
    "plusplus": false,
    "undef": true,
    "unused": false,
    "strict": false,
    "maxparams": 20,
    "maxdepth": 5,
    "maxstatements": 40,
    "maxcomplexity": 11,
    "asi": false,
    "boss": false,
    "debug": false,
    "eqnull": true,
    "esnext": false,
    "evil": false,
    "expr": false,
    "funcscope": false,
    "globalstrict": false,
    "iterator": false,
    "lastsemic": false,
    "laxbreak": false,
    "laxcomma": false,
    "loopfunc": true,
    "maxerr": 50,
    "moz": false,
    "multistr": false,
    "notypeof": false,
    "proto": false,
    "scripturl": false,
    "shadow": false,
    "sub": true,
    "supernew": false,
    "validthis": false,
    "noyield": false,

    "browser": true,
    "node": true,

    "globals": {
        "angular": false,
        "$": false,
        "require": false,
        "global": false,
        "module": false,
        "exports": false,
        "window": false,
        "define": false,
        "FMSGlobals": false,
        "moment": false,
        "console": false,
        "jQuery": false,
        "FileReader": false,
        "_": false
    }
}
```

**Comment by rbatllet** [comment]
Me too! It ignores the .jshintrc global object

**Comment by vernak2539** [comment]
When I pass my rc file in via parameter, it is discounted as well. 

```
.pipe(jshint('./gulp/config/.jshintrc-fe'))
```

**Comment by estreen** [comment]
I did a bit more testing and it looks like the "globals" part of the config is only applied to the first linted file. Other aspects of the config is applied to all files (tested with "undef" for example). So having three files a.js, b.js and c.js with identical content, only a.js get the configured globals. If a.js is renamed to d.js the new first file b.js get the globals applied and the other not.

**Comment by vankasteelj** [comment]
config:
```{
    // Environments
    "browser"  : true,
    "node"     : true,
    "jquery"   : true,
    "devel"    : true,

    // Options
    "curly"    : true,
    "bitwise"  : false,
    "immed"    : true,
    "latedef"  : true,
    "nonbsp"   : true,
    "nonew"    : true,
    "validthis": true,
    "debug"    : true,
    "boss"     : true,
    "eqeqeq"   : true,
    "expr"     : true,
    "eqnull"   : true,
    "quotmark" : "single",
    "trailing" : true,
    "sub"      : true,
    "trailing" : true,
    "undef"    : true,
    "laxbreak" : true,
    "loopfunc" : true,
    "indent"   : 4,
    "newcap"   : false,
    "esversion" : 6,

    // Globals
    "globals": { <a list of "variableName": true,> }
}```

**Comment by rbatllet** [comment]
All my global are ignored

**Comment by vankasteelj** [comment]
I notice that the only visible change you made is `through2` from 0.6.x to 2.x.x. Issue might come from there. Or it's from jshint directly (?)

**Comment by spalger** [comment]
I'm working on this right now folks, sorry for the issue. Thanks for the helpful info though.

If you happen to get a consistently failing test case together please share

**Comment by spalger** [comment]
Alright, I've tracked it down

**Comment by spalger** [comment]
Alright, jshint `v2.0.4' released which should fix the issue.

**Comment by vankasteelj** [comment]
testing now

**Comment by estreen** [comment]
The fix looks good, 2.0.4 works. Thanks for the quick turnaround!
